### PR TITLE
feat: [FFM-4430]: add Setup GitSync redirect button to Flag & Target management pages

### PR DESCRIPTION
**Pull request by Kira-Jennings**
### Summary

add Setup GitSync redirect button to Flag & Target management pages

#### Screenshots
image
image
image
image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cf/strings/strings.en.yaml [hunk 1]
```diff
@@ -155,6 +155,7 @@ featureFlags:
   changeOrReset: Change your filter criteria or reset
   resetFilters: Reset Filters
   clearSearch: Clear Search
+  setupGitSync: Setup Git Sync
```
Comment (crystal-fong) [inline src/modules/75-cf/strings/strings.en.yaml:158]: just wanna check if this is the noun `Setup` or the verb `Set up`

File: src/modules/75-cf/components/FlagActivation/FlagActivationDetails.tsx [hunk 2]
```diff
@@ -128,7 +128,7 @@ const VariationsList: React.FC<{
 }
 
 const FlagActivationDetails: React.FC<FlagActivationDetailsProps> = props => {
-  const { featureFlag, refetchFlag, gitSyncActionsComponent, gitSync, setGovernanceMetadata } = props
+  const { featureFlag, refetchFlag, gitSync, setGovernanceMetadata } = props
```
Comment (chrisblakely01) [inline src/modules/75-cf/components/FlagActivation/FlagActivationDetails.tsx:131]: Think this is fine just FYI theres a GitSyncContext you can get Git stuff from, might save on prop drilling depending on your use case

File: src/modules/75-cf/components/TargetManagementFlagConfigurationPanel/TargetManagementFlagConfigurationPanel.tsx [hunk 3]
```diff
@@ -172,7 +169,7 @@ const TargetManagementFlagConfigurationPanel: FC<TargetManagementFlagConfigurati
                 planEnforcementProps={planEnforcementProps}
                 title={addFlagsDialogTitle}
               />
-              {isGitSyncActionsEnabled && <TargetManagementToolbar />}
```
Comment (chrisblakely01) [inline src/modules/75-cf/components/TargetManagementFlagConfigurationPanel/TargetManagementFlagConfigurationPanel.tsx:175]: Did ya mean to remove this? The use case here is we dont display the Git Stuff if they haven't enabled Git for a project (maybe its changed/moved though)

File: src/modules/75-cf/components/TargetManagementFlagConfigurationPanel/__tests__/TargetManagementFlagConfigurationPanel.test.tsx [hunk 4]
```diff
@@ -328,20 +328,4 @@ describe('TargetManagementFlagConfigurationPanel', () => {
 
     await waitFor(() => expect(screen.queryByText(addFlagsDialogTitle)).not.toBeInTheDocument())
   })
-
-  describe('gitSync', () => {
```
Comment (chrisblakely01) [inline src/modules/75-cf/components/TargetManagementFlagConfigurationPanel/__tests__/TargetManagementFlagConfigurationPanel.test.tsx:332]: guess same Q here as above lol

File: src/modules/75-cf/components/TargetManagementFlagConfigurationPanel/TargetManagementFlagConfigurationPanel.tsx (hunk 3 above)
Comment (Kira-Jennings) [reply src/modules/75-cf/components/TargetManagementFlagConfigurationPanel/TargetManagementFlagConfigurationPanel.tsx:175]: Yes, I updated the GitSyncActions component to output either the Repo/branch button or the Setup Git Sync button, depending if Git has been set up or not (see screenshots). However I just realised I'm only checking for `isGitSyncEnabled` - should the Repo/branch button only be shown if `isGitSyncActionsEnabled` not just `isGitSyncEnabled`?

File: src/modules/75-cf/strings/strings.en.yaml (hunk 1 above)
Comment (Kira-Jennings) [reply src/modules/75-cf/strings/strings.en.yaml]: Checked with Clare and it should be Set Up, good catch!

**Review by chrisblakely01 — APPROVED** [review]
LGTM

**Review by crystal-fong — APPROVED** [review]

**Pull request merged**